include/seqan3/io/alignment_file/input.hpp

```diff
@@ -0,0 +1,1140 @@
+                         selected_field_ids const &) = delete;
+    //!\endcond
+    //!\}
+
+    /*!\name Range interface
+     * \brief Provides functions for record based reading of the file.
+     * \{
+     */
+    /*!\brief Returns an iterator to current position in the file.
+     * \returns An iterator pointing to the current position in the file.
+     *
+     * Equals end() if the file is at end.
+     *
+     * \snippet test/snippet/io/alignment_file/alignment_file_input.cpp begin_and_front
+     *
+     * ### Complexity
+     *
+     * Constant.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    iterator begin() noexcept
+    {
+        return {*this};
+    }
+
+    /*!\brief Returns a sentinel for comparison with iterator.
+     * \returns Iterator to the first element.
+     *
+     * This element acts as a placeholder; attempting to dereference it results in undefined behaviour.
+     *
+     * ### Complexity
+     *
+     * Constant.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    sentinel end() noexcept
+    {
+        return {};
+    }
+
+    /*!\brief Return the record we are currently at in the file.
+     * \returns A reference to the currently buffered record.
+     *
+     * This function returns a reference to the currently buffered record, it is identical to dereferencing begin(),
+     * and begin also always points to the current record on single pass input ranges:
+     *
+     * \snippet test/snippet/io/alignment_file/alignment_file_input.cpp begin_and_front
+     *
+     * In most situations using the iterator interface or a range-based for-loop are preferable to using front(),
+     * because you can only move to the next record via the iterator.
+     *
+     * In any case, don't forget the reference! If you want to save the data from the record elsewhere, use move:
+     *
+     * \snippet test/snippet/io/alignment_file/alignment_file_input.cpp front
+     *
+     * ### Complexity
+     *
+     * Constant.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    reference front() noexcept
+    {
+        return record_buffer;
+    }
+    //!\}
+
+    //!\brief The options are public and its members can be set directly.
+    alignment_file_input_options<typename traits_type::sequence_legal_alphabet> options;
+
+    /*!\brief Access the file's header.
+     *
+     * \details
+     *
+     * You can access the header directly after the construction **with reference information** of the file object.
+     *
+     * ### Example
+     *
+     * \snippet test/snippet/io/alignment_file/alignment_file_input.cpp sam_file
+     *
+     * \snippet test/snippet/io/alignment_file/alignment_file_input.cpp get_header
+     *
+     * \sa seqan3::alignment_file_header
+     */
+    header_type & header()
+    {
+        return *header_ptr;
+    }
+
+protected:
+    //!\privatesection
+
+    //!/brief
```

Comment:
👍

Reply:
also you could give it an optional format parameter and `if constexpr` the set_format inside so all of the constructors can use this,...